### PR TITLE
Create new v1 Campaign PUT Rest Endpoint

### PR DESCRIPTION
**Pull request by shangguanwang**
## Description
Create new Rest endpoint for **PUT** `/v1/campaigns/:id`; Updates the specified campaign when given a JSON payload of modified fields

## Acceptance Criteria
- [x] The PUT endpoint should work properly to update a specific campaign when given a JSON payload of modified fields
- [x] Testing script work as expected - all 3 tests passed

## Related Issue
#615 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Jest Testing
- From your terminal, pull down this branch with `git pull origin 615-api-create-campaign-put-endpoint-Shangguan` and check that branch out with `git checkout 615-api-create-campaign-put-endpoint-Shangguan`
- Run `npm test -- --testPathPattern=campaign_put_endpoint.test.js` in terminal to run the test script

Expect to see all 3 test cases pass: 
test case pass

### Manual Testing in Postman
- Run `npm run dev` to launch the app
**Scenario 1**: Successfully update a campaign
In Postman, first GET `http://localhost:8080/api/v1/campaigns/5`
original campaign 5 data
Create a new PUT request, and input updated campaign JSON under "Body", expect to see a status 200 with the updated campaign as returned result
success put request

**Scenario 2**: The Campaign URL is incorrect
In Postman, PUT `http://localhost:8080/api/v1/campaigns/10` as an example, expect to see a `404 not found` error
campaign not found

**Scenario 3**: Enter invalid input
In Postman, PUT `{
    "campaign": {
        "type": "Random"
    }
}` and expect to see a 500 error
Screen Shot 2023-11-05 at 8 45 38 PM

### CODE REVIEW

File: server/middleware/example.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+function exampleMiddleware(req, res, next) {
```
Comment (DietBepis1) [inline server/middleware/example.js:1]: No need to keep the examples 🙂

File: server/routes/api/v1/campaigns/index.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    return res.status(200).json({ campaigns }).end()
+  } catch (error) {
+    console.error(error.message)
+
+    return res.status(500).json({ msg: 'There was a server error.' }).end()
+  }
+})
+
+// PUT request to update a campaign
```
Comment (DietBepis1) [inline server/routes/api/v1/campaigns/index.js:39]: Looks good!

File: server/routes/api/v1/campaigns/index.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+const express = require('express')
+const Campaign = require('../../../../db/models/campaign')
+
+const router = express.Router()
+
+// Middleware for authentication can get passed into v1 route aggregator
+// so we will assume here that someone requesting these endpoints has already
+// been authorized.
+router.get('/:id', async (req, res) => {
```
Comment (DietBepis1) [inline server/routes/api/v1/campaigns/index.js:9]: You can get rid of these examples also.

**Review by DietBepis1 — CHANGES_REQUESTED** [review]
Good effort! Just a little cleanup and rebasing is needed.

**Comment by shangguanwang** [comment]
> Good effort! Just a little cleanup and rebasing is needed.

Thank you Glenn @DietBepis1 ! I've did the rebase and cleaned up the example code, let me know if there are anything else you want me to change before merge

**Review by DietBepis1 — APPROVED** [review]

**Pull request merged**